preliz/internal/predictive_helper.py

```diff
@@ -68,3 +69,12 @@ def select_prior_samples(selected, prior_samples, model):
     subsample = {rv: prior_samples[rv][selected] for rv in model.keys()}
 
     return subsample
+
+
+def get_engine(source):
+    source = re.sub(r"#.*$|^#.*$", "", source, flags=re.MULTILINE)
+    if re.search(r"\s*(?:bmb|bambi)\.Model\s*", source):
+        return "bambi"
+    if re.search(r"\s*(?:pm|pymc)\.Model\s*", source):
```

Comment:
This will fail if the user non-standard imports, which may be more common than we are expecting. I think we need a slightly more fancy approach. Something like this 

def get_returned_object_class(func):
    sig = inspect.signature(func)
    default_params = {
        k: (v.default if v.default is not inspect.Parameter.empty else np.nan)
        for k, v in sig.parameters.items()
    }
    return func(**default_params).__class__
    
Maybe the logic of this function should be part of `inspect_source` and should also have the logic to directly return the engine. That is the new `inspect_source`, should return 3 objects source, signature, engine